azure/scope/cluster.go

```diff
@@ -1020,9 +1027,15 @@ func (s *ClusterScope) SetControlPlaneSecurityRules() {
 	if !s.ControlPlaneEnabled() {
 		return
 	}
-	if s.ControlPlaneSubnet().SecurityGroup.SecurityRules == nil {
-		subnet := s.ControlPlaneSubnet()
-		subnet.SecurityGroup.SecurityRules = infrav1.SecurityRules{
+
+	subnet := s.ControlPlaneSubnet()
+
+	if subnet.SecurityGroup.SecurityRules == nil {
+		s.AzureCluster.Spec.NetworkSpec.UpdateControlPlaneSubnet(subnet)
+	}
```

Comment:
Could you please add more context on setting `s.AzureCluster.Spec.NetworkSpec.UpdateControlPlaneSubnet(subnet)` three times in this function ? 
Could we not update it once in the end of the function ?